earth_enterprise/src/scons/getversion.py

```diff
@@ -125,10 +125,24 @@ def _GitPreviousReleaseTag(tailTagName):
     return ''
 
 
+def _GitTagRealCommitIdWindows(tagName):
+    """use shell command to retrieve commit id of where the tag points to (Windows version)"""
+    # for some reason .hexsha was not returning the same id....
+    commitId = os.popen("git rev-list -n 1 \"{0}\"".format(tagName.replace("\"", "\\\""))).read().strip()
+    return commitId
+
+def _GitTagRealCommitIdLinux(tagName):
+    """use shell command to retrieve commit id of where the tag points to (Linux version)"""
+    # for some reason .hexsha was not returning the same id....
+    commitId = os.popen("git rev-list -n 1 '{0}'".format(tagName.replace("'", "'\"'\"'"))).read().strip()
+    return commitId
+
 def _GitTagRealCommitId(tagName):
     """use shell command to retrieve commit id of where the tag points to"""
-    # for some reason .hexsha was not returning the same id....
-    return os.popen("git rev-list -n 1 '{0}'".format(tagName.replace("'", "'\"'\"'"))).read().strip()
+    if os.name is 'nt':
+        return _GitTagRealCommitIdWindows(tagName)
+    else:
+        return _GitTagRealCommitIdLinux(tagName)
```

Comment:
Would this work on MacOS ? May be it's not an issue since there seems to be some [build issues with Portable](https://github.com/google/earthenterprise/wiki/Portable-Server#portable-server-on-mac-os) in Mac

Reply:
I don't have a Mac setup to be able to test this.  We have a disclaimer about MacOS running Portable Server on Mac OS not being tested so I think we're ok.